keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -286,7 +286,7 @@ FlowSchema.
 
 Each FlowSchema has:
 - A boolean test of an authenticated request;
-- A matching priority (default value is 1000);
+- A matching precedence;
```

Comment:
i think it's forcing users to think of a numeric matching-precedence every time they create a flow-schema and it will be challenging user-experience i think. consider a cluster w/ 100-ish flow-schemas defined, it takes some effort to think of a number to insert the new flow-schema, isn't it? 

we can set a default precedence so that a user doesn't have to spend time on reading what's the system preset precedences and existing precedences in a running cluster. the default precedences will be lower than most/all of the system presets, the users are supposed to adjust the precedences when resolving conflicts between rules. similar to how `niceness` works in linux processes, users won't deal with that except for resolving preemptions between processes..

Reply:
I do not think we should expect hundreds of flow schemas.  In the KEP it is documented that we expect tens.

Yes the proposal here forcing the admins to think of ordering because this sort of thing needs that.   The relation between schemas is not like the relation between processes competing for CPU, it is like the relation between firewal rules or iptables rules.  Those things are ordered because they have potentially overlapping matching predicates and very different consequences, so the order matters.  There is no convenience of not thinking about the ordering.